### ISSUE TITLE
linter: unreliable lint errors when using multiple threads

### ISSUE THREAD

**Issue by bartlangelaan** [oxc-project/oxc#19034]
### What version of Oxlint are you using?

1.43.0

### What command did you run?

`oxlint --type-aware --type-check --report-unused-disable-directives --deny-warnings`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": [
    "eslint",
    "typescript",
    "unicorn",
    "react",
    "react-perf",
    "nextjs",
    "oxc",
    "import",
    "node",
    "promise"
  ],
  "jsPlugins": ["eslint-plugin-better-tailwindcss"],
  "rules": {
    "better-tailwindcss/enforce-canonical-classes": "error"
  },
  "settings": {
    "better-tailwindcss": {
      "entryPoint": "app/globals.css"
    }
  }
}

```

### What happened?

If I run the linter normally, it doesn't report any error.

But if I add `--threads=1`, it gives multiple errors:

```
> oxlint --type-aware --type-check --report-unused-disable-directives --deny-warnings --threads=1


  × better-tailwindcss(enforce-canonical-classes): The classes: "h-4, w-4" can be simplified to "size-4".
    ╭─[components/ui/dialog.tsx:47:23]
 46 │       <DialogPrimitive.Close className="ring-offset-background focus:ring-ring data-[state=open]:bg-accent data-[state=open]:text-muted-foreground absolute top-4 right-4 rounded-sm opacity-70 transition-opacity hover:opacity-100 focus:ring-2 focus:ring-offset-2 focus:outline-none disabled:pointer-events-none">
 47 │         <X className="h-4 w-4" />
    ·                       ───
 48 │         <span className="sr-only">Close</span>
    ╰────

  × better-tailwindcss(enforce-canonical-classes): The classes: "h-4, w-4" can be simplified to "size-4".
    ╭─[components/ui/dialog.tsx:47:27]
 46 │       <DialogPrimitive.Close className="ring-offset-background focus:ring-ring data-[state=open]:bg-accent data-[state=open]:text-muted-foreground absolute top-4 right-4 rounded-sm opacity-70 transition-opacity hover:opacity-100 focus:ring-2 focus:ring-offset-2 focus:outline-none disabled:pointer-events-none">
 47 │         <X className="h-4 w-4" />
    ·                           ───
 48 │         <span className="sr-only">Close</span>
    ╰────

Found 0 warnings and 2 errors.
Finished in 1.9s on 48 files with 150 rules using 1 threads.

> oxlint --type-aware --type-check --report-unused-disable-directives --deny-warnings

Found 0 warnings and 0 errors.
Finished in 2.0s on 48 files with 150 rules using 11 threads.
```

I am not entirely sure that it is a bug in oxlint (and not the rule itself), but for me the errors show up consistently single-threaded and don't show up when using multiple threads - which makes me believe that it is something in oxlint.

It happens in this repository: http://github.com/bartlangelaan/techpref.dev/tree/8baa1b3e75e5facdff559354a8b6e37c8683b05c (removed the rules in later commits - but checking out this commit it should still be reproducible)

**Comment by bartlangelaan** [comment]
I also tried this using eslint. 

`pnpm add eslint typescript-eslint`

```
import eslintPluginBetterTailwindcss from "eslint-plugin-better-tailwindcss";
import { defineConfig } from "eslint/config";
import { parser as eslintParserTypeScript } from "typescript-eslint";

export default defineConfig([
  {
    extends: [
      { ...eslintPluginBetterTailwindcss.configs.recommended, rules: {} },
    ],
    rules: {
      "better-tailwindcss/enforce-canonical-classes": ["error"],
    },
    settings: {
      "better-tailwindcss": {
        entryPoint: "app/globals.css",
      },
    },
  },
  {
    files: ["**/*.{ts,tsx,cts,mts}"],
    languageOptions: {
      parser: eslintParserTypeScript,
      parserOptions: {
        project: true,
      },
    },
  },
  {
    files: ["**/*.{jsx,tsx}"],
    languageOptions: {
      parserOptions: {
        ecmaFeatures: {
          jsx: true,
        },
      },
    },
  },
]);
```

This consistently shows that there are 14 errors - so I think the bug is not in the better-tailwindcss rule.

**Comment by camc314** [comment]
This appears to be a bug in the eslint plugin:

https://github.com/schoero/eslint-plugin-better-tailwindcss/issues/321

**Comment by camc314** [comment]
@overlookmotel do you mind taking a look at this as a quick sanity check for me? I'd like to confirm with you that this isn't a bug in oxlint

**Comment by camc314** [comment]
Closing as fixed - the upstream repo has merged my PR fixing the issue - updateing the plugin version should resolve the inconsistent behaivour.
